### PR TITLE
Implementation of CIFAR10 Image Classifier PyTorch

### PR DESCRIPTION
**Pull request by DDibyajyot**
Hi! Here is my Bootcamp submission for the computer vision subteam at UWARG.
I have implemented a CIFAR10 Image Classifier using PyTorch with a final accuracy of 72.4%. This model was trained for 25 epochs and the plot included is plotted using Pyplot. 
PFA the screenshots of the training screen and the plot. Please let me know if there are any issues or if I need to fix anything!

FinalAcc
TrainingScreen

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -12,6 +12,255 @@
...
+    print('CUDA is available!  Training on GPU ...')
+
+
+# check if CUDA is available
+train_on_gpu = torch.cuda.is_available()
+
+if not train_on_gpu:
+    print('CUDA is not available.  Training on CPU ...')
+else:
+    print('CUDA is available!  Training on GPU ...')
```
Comment (rayjinghaolei) [inline main.py:46]: I think this is repeated code haha

**Review by rayjinghaolei — CHANGES_REQUESTED** [review]
Overall the content is great, in terms of the formatting, things can be done more elegantly.  You can separate your classes into different files, and your preprocessing, data loading, training loop, evaluation loop, and validation, can all be separated into different functions while the main function calls those helper functions to control the overall program

**Comment by DDibyajyot** [comment]
Hi! I have tried to follow your feedback and separated classes into other files as well as converted the code into helper functions which are then called by the main function.
I hope this would be alright, please do let me know if there is anything else I could change. Once again thank you so much for your feedback!
